e2e/src/test.js

```diff
@@ -0,0 +1,171 @@
+// Prerequisites:
+// - npm install
+// - BROWSERSTACK_USER env var
+// - BROWSERSTACK_KEY env var
+// - BROWSERSTACK_APP env var
+// - for local sync server, have BrowserStackLocal running (https://www.browserstack.com/local-testing#command-line)
+
+const wdio = require('webdriverio');
+const fetch = require('node-fetch');
+const moment = require('moment');
+
+const services = require('../fixtures/mobile-services');
+
+const TASKS_TO_CREATE = 100;
+
+const localServer = services.services.find(service => service.url.includes('bs-local.com'));
+
+const opts = {
+  hostname:'hub-cloud.browserstack.com',
+  logLevel: 'error',
+  capabilities: {
+    'os_version': '7.1',
+    'device': 'Google Pixel',
+    'real_mobile': 'true',
+    'project': 'AeroGear DataSync Performance Tests',
+    'name': 'tests',
+    'browserstack.local': localServer ? 'true' : 'false',
+    'browserstack.user': process.env.BROWSERSTACK_USER,
+    'browserstack.key': process.env.BROWSERSTACK_KEY,
+    'app': process.env.BROWSERSTACK_APP,
+    'autoWebview': true,
+    'browserstack.appium_version': '1.9.1',
+    'browserstack.networkLogs': true
+  }
+};
+
+const setNetwork = async (profile, client) => {
+  const buff = Buffer.from(`${process.env.BROWSERSTACK_USER}:${process.env.BROWSERSTACK_KEY}`);
+  await fetch(`https://api-cloud.browserstack.com/app-automate/sessions/${client.sessionId}/update_network.json`, {
+    body: `{"networkProfile":"${profile}"}`,
+    headers: {
+      Authorization: `Basic ${buff.toString('base64')}`,
+      "Content-Type": "application/json"
+    },
+    method: "PUT"
+  });
+};
+
+const getNumOfTasks = async client => {
+  const result = await client.executeAsync(async done => {
+    const { voyeager, queries } =  window.aerogear;
+
+    const result = await voyeager.apolloClient.query({
+      query: queries.GET_TASKS,
+      fetchPolicy: 'network-only',
+      errorPolicy: 'none'
+    });
+
+    done(result.data.allTasks.length);
+  });
+
+  console.log(`number of tasks: ${result}`);
+
+  return result;
+};
+
+const keycloakEnabled = () => {
+  return services.services.find(service => service.type === 'keycloak');
+};
+
+const authenticate = async client => {
+  console.log('authenticating');
+
+  const mainWindow = await client.getWindowHandle();
+  const allWindows = await client.getWindowHandles();
+  const loginPage = allWindows.find(w => w !== mainWindow);
+  await client.switchToWindow(loginPage);
+
+  const usernamEl = await client.$('#username')
+  await usernamEl.setValue('test');
+  
+  const passwordEl = await client.$('#password')
+  await passwordEl.setValue('123');
```

Comment:
wdyt about taking the values for keycloak user&pass from env vars and documenting it?

Reply:
good point, updated